bcda/api/requests.go

```diff
@@ -42,10 +44,57 @@ func init() {
 	db.DB().SetMaxIdleConns(utils.GetEnvInt("BCDA_DB_MAX_IDLE_CONNS", 25))
 	db.DB().SetConnMaxLifetime(time.Duration(utils.GetEnvInt("BCDA_DB_CONN_MAX_LIFETIME_MIN", 5)) * time.Minute)
 	repository := postgres.NewRepository(db)
-	models.GetService(repository, cutoffDuration, utils.GetEnvInt("BCDA_SUPPRESSION_LOOKBACK_DAYS", 60))
+	svc = models.NewService(repository, cutoffDuration, utils.GetEnvInt("BCDA_SUPPRESSION_LOOKBACK_DAYS", 60))
 }
 
-func BulkRequest(resourceTypes []string, w http.ResponseWriter, r *http.Request, retrieveNewBeneHistData bool) {
+func BulkPatientRequest(w http.ResponseWriter, r *http.Request) {
+	resourceTypes, err := ValidateRequest(r)
+	if err != nil {
+		responseutils.WriteError(err, w, http.StatusBadRequest)
+		return
+	}
+	reqType := models.DefaultRequest // historical data for new beneficiaries will not be retrieved (this capability is only available with /Group)
+	bulkRequest(resourceTypes, w, r, reqType)
+}
+
+func BulkGroupRequest(w http.ResponseWriter, r *http.Request) {
+	const (
+		groupAll    = "all"
+		groupRunout = "runout"
+	)
+
+	reqType := models.DefaultRequest
+	groupID := chi.URLParam(r, "groupId")
+	switch groupID {
+	case groupAll:
+		// Set flag to retrieve new beneficiaries' historical data if _since param is provided and feature is turned on
+		_, ok := r.URL.Query()["_since"]
+		if ok && utils.GetEnvBool("BCDA_ENABLE_NEW_GROUP", false) {
+			reqType = models.RetrieveNewBeneHistData
+		}
+	case groupRunout:
+		if !utils.GetEnvBool("BCDA_ENABLE_RUNOUT", false) {
+			oo := responseutils.CreateOpOutcome(responseutils.Error, responseutils.Exception, responseutils.RequestErr, "Runouts are not enabled")
```

Comment:
Consider either of the following error messages:
(1) Invalid group ID
(2) Runout group is not supported

Reply:
I'll go with `(2)`

Reply:
I agree with that.  If `BCDA_ENABLE_RUNOUT` is turned off, then `runout` is an invalid group ID.